contributions/api/controllers/contributions.py

```diff
@@ -810,25 +810,35 @@ def get_data_list(name, login_id, is_login):
             else:
```

Comment:
Would it be better to add more comments to the get_data_list method? It doesn't seem to be easy to follow. 

Also, for future improvement, there seem to be a few utility methods in /controllers/contributions.py. It may be better to move them to appropriate utility programs that you have already created. Thanks.

Reply:
moved the method to otherutils and added comments. It is very very very complicated as we discussed before because of the 401 and 404 separation. If we simplify that, we could make it much simpler.......